Keen.NetStandard/KeenClient.cs

```diff
@@ -972,5 +973,177 @@ public JObject Query(string queryName, Dictionary<string, string> parms)
+        /// <param name="timeframe">Limits retrieval of results to a specific portion of the Cached Dataset</param>
+        /// <returns>An instance of Newtonsoft.Json.Linq.JObject containing query results and metadata defining the cached dataset.</returns>
+        public JObject QueryDataset(string datasetName, string indexBy, string timeframe)
+        {
+            try
+            {
+                return QueryDatasetAsync(datasetName, indexBy, timeframe).Result;
+            }
+            catch (AggregateException ex)
+            {
+                throw ex.TryUnwrap();
+            }
+        }
+
+        /// <summary>
+        /// Get the definition of your cached dataset.
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to get the definition of.</param>
+        /// <returns>An instance of Keen.Core.Dataset.DatasetDefinition containing metadata about your cached dataset.</returns>
+        public async Task<DatasetDefinition> GetDatasetDefinitionAsync(string datasetName)
+        {
+            return await Datasets.GetDefinitionAsync(datasetName);
+        }
+
+        /// <summary>
+        /// Get the definition of your cached dataset.
+        /// </summary>
+        /// <param name="datasetName">Name of cached dataset to get the definition of.</param>
+        /// <returns>An instance of Keen.Core.Dataset.DatasetDefinition containing metadata about your cached dataset.</returns>
+        public DatasetDefinition GetDatasetDefinition(string datasetName)
+        {
+            try
+            {
+                return GetDatasetDefinitionAsync(datasetName).Result;
+            }
+            catch (AggregateException ex)
+            {
+                throw ex.TryUnwrap();
+            }
+        }
+
+        /// <summary>
+        /// Lists the first n cached dataset definitions in your project.
+        /// </summary>
+        /// <param name="limit">How many cached dataset definitions to return at a time (1-100). Defaults to 10.</param>
+        /// <param name="afterName">A cursor for use in pagination. afterName is the Cached Dataset name that defines your place in the list. 
+        /// For instance, if you make a list request and receive 100 Cached Dataset definitions, 
+        /// ending with dataset_foo you can use dataset_foo as your afterName to retrieve the next page of definitions. 
+        /// Lists also return with helper “NextPageUrl” that uses AfterName, 
+        /// so your subsequent call can fetch the next page of the list easily.</param>
+        /// <returns>An instance of Keen.Core.Dataset.DatasetDefinitionCollection containing the total count, next page url and list of DatasetDefinitions.</returns>
+        public Task<DatasetDefinitionCollection> ListDatasetDefinitionsAsync(int limit = 10, string afterName = null)
+        {
+            return Datasets.ListDefinitionsAsync(limit, afterName);
```

Comment:
Another reason not to have the Datasets object is the extra glue code needed to call into it. The implementation could be here or in a partial class.